CMakeLists.txt

```diff
@@ -147,6 +147,18 @@ add_custom_target(
 add_dependencies(symlinks videoencoder snd2png debugviewer tinycv)
 
 # create installable versions of isotovideo and cv.pm
+set(OS_AUTOINST_VERSION "" CACHE STRING "os-autoinst version to set for the installation")
+if (OS_AUTOINST_VERSION)
+    set(OS_AUTINST_VERSION_REPLACE "${OS_AUTOINST_VERSION}")
+elseif (GIT_BIN AND IS_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/.git")
+    find_program(GIT_BIN git)
+    execute_process(COMMAND "${GIT_BIN}" rev-parse HEAD
+                    WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
+                    OUTPUT_VARIABLE OS_AUTINST_VERSION_REPLACE)
+    string(REPLACE "\n" "" OS_AUTINST_VERSION_REPLACE "${OS_AUTINST_VERSION_REPLACE}")
```

Comment:
That's actually not "moving the replacement to CMake", it's something additional. Why care about calling git dynamically here as well? That's a bit duplicating what isotovideo is doing internally. I understand that you want to cover the additional use case of "user installs from a git checkout and wants to have a valid version included" but do we need that?

Reply:
> Why care about calling git dynamically here as well?

We came to this after noticing bad user experience: A user ended up with an installation that complained not being within a Git directory. Just doing the replacement if `OS_AUTINST_VERSION` is defined (with no fallback) would not improve that. Note that `git rev-parse HEAD` is not too much logic in my opinion and we also don't *need* to keep this in sync with isotovideo itself. This is merely to have good enough fallback.

Reply:
I changed the commit message to make the real intention more clear.